File: leetcode_75/DP_1D/python/746.py
from typing import List


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        cost.append(0)
        dp = [0] * (n + 2)
        dp[1] = cost[0]

        for i in range(2, n + 2):
            dp[i] = min(dp[i - 1], dp[i - 2]) + cost[i - 1]

        return dp[-1]

File: leetcode_75/DP_1D/cpp/746.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int minCostClimbingStairs(vector<int> &cost)
    {
        int n = cost.size();
        cost.push_back(0);

        vector<int> dp(n + 2, 0);
        dp[1] = cost[0];

        for (int i = 2; i < n + 2; ++i)
        {
            dp[i] = min(dp[i - 1], dp[i - 2]) + cost[i - 1];
        }

        return dp[n + 1];
    }
};